Output NuGet packages to a common artifact directory when building locally

drub0y opened the pull request:
Fixes #391.

tomlm commented:
2 issues
1. This breaks our server build signing scripts. I tried updating them with your branch but it's not working and I don't have to time to mess with them right now.  
2. new dialog library is missing

drub0y commented:
@tomlm 

1. I'll take a peek over at the build to see what's up.
2. IIRC, at the time I did this Dialogs didn't actually have a nupkg configured yet. I'll rebase this against latest and get that dialed in too.

Stand by....

drub0y commented:
Ok, rebased against `master` to pick up you having added NuGet packaging to the Dialogs project and added the `PackageOutputPath` to the Dialgos project now as well, so that should take care of `2`.

@tomlm Can you tell me which build in VSTS is the one you use to cut the official packages? Is it the one named `Bot.Builder-DotNet-CI`? I see `Bot.Builder-DotNet-Pull Request` also, but that's obviously the one triggered by these PRs and it's working.

drub0y commented:
@tomlm can you spare a few seconds so we can get this closed out?
